brewtils/plugin.py

```diff
@@ -307,89 +242,90 @@ def _initialize_system(self):
                 # the create can return a conflict. In that case just try the get again
-                existing_system = self.bm_client.find_unique_system(
-                    name=self.system.name, version=self.system.version
+                existing_system = self._ez_client.find_unique_system(
+                    name=self._system.name, version=self._system.version
                 )
 
         # If we STILL can't find a system something is really wrong
         if not existing_system:
             raise PluginValidationError(
                 "Unable to find or create system {0}-{1}".format(
-                    self.system.name, self.system.version
+                    self._system.name, self._system.version
                 )
             )
 
         # We always update with these fields
         update_kwargs = {
-            "new_commands": self.system.commands,
-            "metadata": self.system.metadata,
-            "description": self.system.description,
-            "display_name": self.system.display_name,
-            "icon_name": self.system.icon_name,
+            "new_commands": self._system.commands,
+            "metadata": self._system.metadata,
+            "description": self._system.description,
+            "display_name": self._system.display_name,
+            "icon_name": self._system.icon_name,
         }
 
         # And if this particular instance doesn't exist we want to add it
-        if not existing_system.has_instance(self.instance_name):
-            update_kwargs["add_instance"] = Instance(name=self.instance_name)
+        if not existing_system.has_instance(self.config.instance_name):
+            update_kwargs["add_instance"] = Instance(name=self.config.instance_name)
 
-        return self.bm_client.update_system(existing_system.id, **update_kwargs)
+        return self._ez_client.update_system(existing_system.id, **update_kwargs)
 
     def _initialize_instance(self):
         # Sanity check to make sure an instance with this name was registered
-        if not self.system.has_instance(self.instance_name):
+        if not self._system.has_instance(self.config.instance_name):
             raise PluginValidationError(
-                'Unable to find registered instance with name "%s"' % self.instance_name
+                "Unable to find registered instance with name '%s'"
+                % self.config.instance_name
             )
 
-        return self.bm_client.initialize_instance(
-            self.system.get_instance(self.instance_name).id
+        return self._ez_client.initialize_instance(
+            self._system.get_instance(self.config.instance_name).id
         )
 
     def _initialize_processors(self):
         """Create RequestProcessors for the admin and request queues"""
         # If the queue connection is TLS we need to update connection params with
         # values specified at plugin creation
-        connection_info = self.instance.queue_info["connection"]
```

Comment:
The function name doesn't match up with what is going on here. This initializes the rabbitmq admin/request queues. If we ever decided to provide a different connection vehicle between BG and Plugins, then this will need to be scalable to different types. If we want to leave it as only RabbitMQ for v3, then we can leave this alone.